docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,240 @@
+- *A translation asset* &ndash; A JavaScript file or part of the file with generated translations for one language.
+
+## Writing a localizable UI
+
+All *messages* that need localization should be passed to the special CKEditor 5's {@link module:utils/locale~Locale#t `t()` function}. This function can be retrieved from the editor's {@link module:utils/locale~Locale} instance: `const { t } = editor.locale;` or from any view method: `const t = this.t;`.
+
+As the first argument, the `t()` function accepts either a string literal or an object literal containing the `id`, `string` and `plural` (optional) properties. The string literal will serve as both the *message ID* and the *message string*.
+
+As the second argument, the translation function accepts a value or an array of values. These values will be used to fill the placeholders in more advanced translation scenarios. If the `plural` property is specified, the first value will be used as the quantity determining the plural form.
+
+When using the `t()` function, you can create your own *localizable messages* or reuse *messages* created in CKEditor 5 packages that your project depends on. In case of reusing *messages*, you will not need to worry about translating them as all work will be done by the CKEditor 5 team and [Transifex translators](https://www.transifex.com/ckeditor/ckeditor5/) anyway. Obviously, your help in translating will still be appreciated!
+
+For simple *localizable messages*, use the string form for simplicity:
+
+```js
+const emojiName = 'cat';
+
+// Assuming that the English language was picked:
+t( 'insert emoji' ); // "insert emoji"
+t( 'insert %0 emoji', emojiName ); // "insert cat emoji"
+t( 'insert %0 emoji', [ emojiName ] ); // "insert cat emoji"
+```
+
+For more advanced scenarios, use plain object forms:
+
+```js
+const quantity = 3;
+
+// Assuming that the English language was picked:
+t( { string: '%0 emoji', id: 'INSERT_EMOJI' }, 'insert' ); // "insert emoji"
+t( { string: '%0 emoji', plural: '%0 emojis', id: 'EMOJI' }, quantity ); // "3 emojis"
+t( { string: '%1 %0 emoji', plural: '%1 %0 emojis', id: 'INSERT_EMOJIS' }, [ quantity, 'Insert' ] ); // "Insert 3 emojis"
+```
+
+<info-box warning>
+	Note that the `t()` function should not be used as a method of the `Locale` class because during the translation process a static code analyzer catches *localizable messages* only from function forms. For the same reason its first argument accepts only a string literal or an object literal. `Locale#t()` method calls and incompatible expressions passed to the `t()` calls will produce build-time warnings.
+</info-box>
+
+### Example: Localizing the plugin UI
+
+This example shows how to create a localizable user interface of a plugin.
+
+```js
+// ...
+editor.ui.componentFactory.add( 'smilingFaceEmoji', locale => {
+	const command = editor.commands.get( 'insertSmilingFaceEmoji' );
```

Comment:
AFAICS this variable is not used anywhere in the code. You might have wanted to link it to the button state but this is not needed in this example, I think.

Reply:
You're right, I simplified the example and I forgot to remove it afterward.